backend/controller/paymentController.js

```diff
@@ -47,36 +56,66 @@ const createSubscription = asyncHandler(async (req, res, next) => {
     return next(new customError("Admin cannot purchase a subscription", 400));
   }
 
-  // FIXME: Below task is async, wrap it in trycatch to handle any potential errors
   // Creating a subscription using razorpay that we imported from the config/rasorpayConfig.js
-  const subscription = await razorpay.subscriptions.create({
-    plan_id: planID, // The unique plan ID
-    customer_notify: 1, // 1 means razorpay will handle notifying the customer, 0 means we will not notify the customer
-    total_count: 1, // 1 means it will charge only one month sub.
-  });
-
-  // FIXME: Make sure subscription did not return any result and then save the details in DB
-  // Adding the ID and the status to the user account
-  user.subscription.id = subscription.id;
-  user.subscription.status = subscription.status;
+  try {
+    const subscription = await razorpay.subscriptions.create({
+      plan_id: "planID", // The unique plan ID
+      customer_notify: 1, // 1 means razorpay will handle notifying the customer, 0 means we will not notify the customer
+      total_count: 1, // 1 means it will charge only one month sub.
+    });
+
+    // Adding the ID and the status to the user account
+    user.subscription.id = subscription.id;
+    user.subscription.status = subscription.status;
+  } catch (error) {
+    return next(
+      new customError(
+        error.error.description ||
+          "Error during creating RazorPay subscription",
+        400
+      )
+    );
+  }
 
   // Saving the user object
   await user.save();
 
   res.status(200).json({
+    statusCode: 200,
```

Comment:
Looks like this should be a 201, since it is creating something.

Reply:
added 201